go/vt/vtgateproxy/discovery.go

```diff
@@ -258,8 +276,22 @@ func (b *JSONGateResolverBuilder) parse() (bool, error) {
 			return false, fmt.Errorf("error parsing JSON discovery file %s: port field %s has invalid value %v", b.jsonPath, b.portField, port)
 		}
 
-		targets = append(targets, targetHost{fmt.Sprintf("%s:%s", address, port), poolType.(string), affinity.(string)})
+		target := targetHost{fmt.Sprintf("%s:%s", address, port), poolType.(string), affinity.(string)}
+		targets[target.PoolType] = append(targets[target.PoolType], target)
+	}
+
+	for poolType := range targets {
+		if b.affinityField != "" {
+			sort.Slice(targets[poolType], func(i, j int) bool {
+				return b.affinityValue == targets[poolType][i].Affinity
+			})
+		}
+		if len(targets[poolType]) > *numConnections {
```

Comment:
Much simpler than my unnecessary `min()` function.

Reply:
we could use the built in `min` if we weren't on go 1.18 😢

Reply:
I know!!